src/EventListener/CacheControlListener.php

```diff
@@ -143,7 +143,11 @@ public function onKernelResponse(FilterResponseEvent $event)
         if (!empty($options['etag'])
             && ($options['overwrite'] || null === $response->getEtag())
         ) {
-            $response->setEtag(md5($response->getContent()));
+            if ($options['etag'] === 'weak') {
+                $response->setEtag(md5($response->getContent()), true);
```

Comment:
lets not do an if-else but simplify to `$response->setEtag(md5(...), 'weak' === $options['etag']);`